pkg/core/blockchain.go

```diff
@@ -699,6 +699,17 @@ func (bc *Blockchain) storeBlock(block *block.Block, txpool *mempool.Pool) error
 	}
 
 	bc.lock.Lock()
+	if bc.config.RemoveUntraceableBlocks {
```

Comment:
It doesn't need to be locked certainly (as long as removal is atomic for block).

Reply:
Fixed.

Reply:
Have you tried async version? 5% overhead is not a lot, but still is something.